ServerAPI/Controllers/GameController.cs

```diff
@@ -0,0 +1,61 @@
+using Microsoft.AspNetCore.Http.HttpResults;
+using Microsoft.AspNetCore.Mvc;
+
+namespace ServerAP.Controllers;
+[ApiController]
+[Route("GameController")]
+
+public class GameController : ControllerBase
+{
+    private readonly Game _game;
+    private readonly PollingHandler _handler;
+
+    public GameController(GameDispenser gameDispenser,PollingHandler pollingHandler)
+    {
+        _game = gameDispenser.DispenseGame();
+        _handler = pollingHandler;
+    }
+    
+    [HttpGet("WaitForTurn")]
+    public async Task<IActionResult> WaitForTurn()
+    {
+        while (!_handler.Notified)
+        {
+            await Task.Delay(1000);
+        }
+        
+        return Ok(_handler.Consume());
+    }
+    
+    [HttpPost( "MakeTurn")]
+    public  IActionResult MakeTurn(TurnInfo turnInfo)
+    {
+        
+        _game.TurnInfo = turnInfo;
+        
+        var turnValidator = _game.ValidateTurn();
```

Comment:
you can deconstruct the tuple, so you can have more neat names
e.g. `(var isValid, var errorMessage) = _game.ValidateTurn();`